internal/common/baseclient/iigo.go

```diff
@@ -53,6 +53,13 @@ func (c *BaseClient) GetTaxContribution() shared.Resources {
 	return 0
 }
 
+// GetCommonPoolContribution requests a client resource contribution towards the common pool.
+// This is used for disaster mitigation in the event of a disaster.
+// OPTIONAL: Use this method to contribue towards CP
+func (c *BaseClient) GetCommonPoolContribution() shared.Resources {
+	return 0
+}
```

Comment:
I think that this is unnecessary, there is already a function called GetTaxContribution() which covers contributions to the common pool.

Reply:
Yeah I was aware of this. But thought having a separate function offers more flexibility. Otherwise agents can't not pay tax but still contribute to CP. See my first discussion with Naim above. Open to unifying if there's a strong reason against separate methods but it makes more sense to me tbh - especially since CP contributions are conceptually different to paying tax.

Reply:
Maybe we should check with design on this as I think they conceptually are the same (or similar). 

The point of the tax is to make sure that there is enough in the common pool to pay for IIGO and to mitigate a disaster. The President sets a minimum contribution but islands are free to pay more. Also, if there are 2 separate functions, this implies that an island can contribute to the common pool without paying tax but tax is defined as a minimum contribution so they have in fact paid the tax. Hope that makes sense!

Reply:
What if they want to contribute an amount to CP that is below the min tax amount set by President?

Reply:
To me they don't seem to be conceptually different, a tax is just a rule about the minimum contribution an agent should make. It is always fine to contribute more. With this in mind your argument about agents that can't not pay tax but still contribute to the CP does not make sense - if they contribute enough they have paid the tax.

Reply:
> What if they want to contribute an amount to CP that is below the min tax amount set by President?

That's fine, it's entirely the agent's choice. They just might get caught and sanctioned by the Judge at the next IIGO session.

Reply:
@Spruudis not sure I agree. When would you ever pay more tax than required in a practical case? Also, I was referring specifically to the case where they _don't_ contribute enough to cover base tax amount. Anyway, this is just going to take too long to debate or change so I'm just going to merge the CP contributions into tax as you've suggested.